tests/Integration/goldens/asset/fragments/Google/Cloud/Asset/V1/SearchAllIamPoliciesRequest.build.txt

```diff
@@ -0,0 +1,106 @@
+     *
+     *                               Examples:
+     *
+     *                               * `policy:amy&#64;gmail.com` to find IAM policy bindings that specify user
+     *                               "amy&#64;gmail.com".
+     *                               * `policy:roles/compute.admin` to find IAM policy bindings that specify
+     *                               the Compute Admin role.
+     *                               * `policy:comp*` to find IAM policy bindings that contain "comp" as a
+     *                               prefix of any word in the binding.
+     *                               * `policy.role.permissions:storage.buckets.update` to find IAM policy
+     *                               bindings that specify a role containing "storage.buckets.update"
+     *                               permission. Note that if callers don't have `iam.roles.get` access to a
+     *                               role's included permissions, policy bindings that specify this role will
+     *                               be dropped from the search results.
+     *                               * `policy.role.permissions:upd*` to find IAM policy bindings that specify a
+     *                               role containing "upd" as a prefix of any word in the role permission.
+     *                               Note that if callers don't have `iam.roles.get` access to a role's
+     *                               included permissions, policy bindings that specify this role will be
+     *                               dropped from the search results.
+     *                               * `resource:organizations/123456` to find IAM policy bindings
+     *                               that are set on "organizations/123456".
+     *                               * `resource=//cloudresourcemanager.googleapis.com/projects/myproject` to
+     *                               find IAM policy bindings that are set on the project named "myproject".
+     *                               * `Important` to find IAM policy bindings that contain "Important" as a
+     *                               word in any of the searchable fields (except for the included
+     *                               permissions).
+     *                               * `resource:(instance1 OR instance2) policy:amy` to find
+     *                               IAM policy bindings that are set on resources "instance1" or
+     *                               "instance2" and also specify user "amy".
+     *                               * `roles:roles/compute.admin` to find IAM policy bindings that specify the
+     *                               Compute Admin role.
+     *                               * `memberTypes:user` to find IAM policy bindings that contain the "user"
+     *                               member type.
+     * @param mixed  $optionalFields {
```

Comment:
I am thinking it may make more sense to not include the optional fields in the builders. This would encourage the use of fluid setters as opposed to a big array of properties:

```php
// using fluid setters
$request.= SearchAllIamPoliciesRequest::build('scope', 'query')
    ->setAssetTypes($assetTypes);
// using array of "optionalFields"
$request.= SearchAllIamPoliciesRequest::build('scope', 'query', ['asset_types' => $assetTypes]);

```

Reply:
That is fine with me, and those fields can still be set after the request is created. The primary use case for these builders/`method_signature` is to name the bare minimum of fields necessary to make a successful request. SGTM.

Reply:
Done!